lib/dry/validation/schema.rb

```diff
@@ -92,6 +94,14 @@ def [](name)
       def predicates
         self.class.predicates
       end
+
+      private
+
+      def validate_input(input)
+        unless input.is_a?(Hash)
+          raise ArgumentError, "expected a Hash but received +#{input.inspect}+"
+        end
```

Comment:
If you want to be very defensive (and it can happen in some cases) it should be able to do something like this:

``` ruby
class Schema < Dry::Validation::Schema
  hash? do
    key(:name, &:filled?)
  end
end
```

Lemme know if that works (it should)